src/test/java/BaseTest.java

```diff
@@ -1,11 +1,55 @@
+
+
 import io.github.bonigarcia.wdm.WebDriverManager;
-import org.openqa.selenium.chrome.ChromeOptions;
-import org.testng.annotations.BeforeSuite;
+        import org.openqa.selenium.chrome.ChromeDriver;
+        import org.openqa.selenium.chrome.ChromeOptions;
+        import org.testng.annotations.BeforeMethod;
+        import org.testng.annotations.BeforeSuite;
+        import org.testng.annotations.Parameters;
+
+        import java.time.Duration;
 
 public class BaseTest {
 
     @BeforeSuite
     static void setupClass() {
         WebDriverManager.chromedriver().setup();
     }
-}
\ No newline at end of file
+
+    @BeforeMethod
+    @Parameters({"BaseURL"})
+    public void launchBrowser(String BaseURL) {
+        ChromeOptions options = new ChromeOptions();
+        options.addArguments("--remote-allow-orgins=*");
+
+        ChromeDriver driver;
+        driver = new ChromeDriver(options);
+        driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+
+       wait = new WebDriverWait(driver, Duration.ofSeconds(10));
```

Comment:
Hello,
Your homework 20 has been reviewed, but the test script execute failed. Please recheck and correct it.

//HINT:
1. In the BaseTest have many error, frist is missing import library. 
At line 29 initialization wait but you not delarce type for variable wait. The same with variable url below.
![Uploading image.png…]()